californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -1070,7 +1092,7 @@ public void cancelObservation(Token token) {
 	 *
 	 * @param task the task
 	 */
-	private void runInProtocolStage(final Runnable task) {
+	protected void runInProtocolStage(final Runnable task) {
```

Comment:
The most scope changes seems to be not used in `OSCoreEndpoint`.

Maybe, if we offer a better way to create a special `CoapStack`, subclassing may be obsolete at all.

Reply:
> A lot of variables needs to be accessible for the OSCoreEndpoint since the CoapEndpoint's constructor has become very strict. 

I can't see, where you access these variables, can you point me to that?
You access `this.coapstack` but the others seems to be not used by `OSCoreEndpoint`.

The number of CoapEndpoint constructor has grown over the time by adding optional parameters. Therefore I changed it to a "Builder2, which unfortunately, is not sub-class friendly. But your constructors can simply call "this()" and so only one constructor must then call "super()" and use the "null"-defaults. 

> If you brought back one of these early constructors which don't need a crazy amount of input the exposure for CoapEndpoint would not need to be changed as much.

I don't see, that this is related. But maybe, you can point me to the usage of these variables in OSCORE.